development/tools/cmd/gcscleaner/main.go

```diff
@@ -3,88 +3,112 @@ package main
 import (
 	"context"
 	"flag"
+	_ "math/rand"
 	"regexp"
 	"strings"
+	"syscall"
 	"time"
 
-	"cloud.google.com/go/storage"
 	"github.com/kyma-project/test-infra/development/tools/pkg/gcscleaner"
+	"github.com/kyma-project/test-infra/development/tools/pkg/gcscleaner/storage"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
-// Config structure aggregating application configuration arguments
-type options struct {
-	ProjectName            string
-	BucketLifespanDuration time.Duration
-	ExcludedBucketNames    []string
-	DryRun                 bool
-	BucketNameRegexp       regexp.Regexp
+// Config cleaner configuration
+type Config struct {
+	ProjectName               string
+	BucketLifespanDuration    time.Duration
+	ExcludedBucketNames       []string
+	IsDryRun                  bool
+	BucketNameRegexp          regexp.Regexp
+	BucketObjectWorkersNumber int
+	LogLevel                  logrus.Level
 }
 
-func main() {
-	ctx := context.Background()
+var (
+	argProjectName                  string
+	argExcludedBucketNames          string
+	argBucketLifespanDuration       string
+	argBucketNameRegexp             string
+	argDryRun                       bool
+	argBucketObjectWorkerNumber     int
+	argLogLevel                     string
+	bucketLifespanDurationDefault   = "2h"
+	bucketObjectWorkerNumberDefault = 1
+	logLevelDefault                 = "info"
+	// ErrInvalidProjectName returned if project name argument is invalid
+	ErrInvalidProjectName = errors.New("invalid project name argument")
+	// ErrInvalidDuration returned if duration argument is invalid
+	ErrInvalidDuration = errors.New("invalid duration argument")
+	// ErrEmptyBucketNameRegexp returned if argBucketNameRegexp is empty
+	ErrEmptyBucketNameRegexp = errors.New("empty bucketNameRegexp argument")
+	// ErrInvalidBucketNameRegexp returned if argBucketNameRegexp is invalid
+	ErrInvalidBucketNameRegexp = errors.New("invalid bucketNameRegexp argument")
+	// ErrInvalidLogLevel returned when argLogLevel is invalid
+	ErrInvalidLogLevel = errors.New("bucket object has invalid attributes")
+)
 
-	options, err := readOptions()
+func main() {
+	rootCtx := context.Background()
+	cfg, err := readCfg()
 	if err != nil {
 		logrus.Fatal(errors.Wrap(err, "reading arguments"))
 	}
-
-	client, err := storage.NewClient(ctx)
+	logrus.SetLevel(cfg.LogLevel)
+	client, err := storage.NewClient(rootCtx)
 	if err != nil {
 		logrus.Fatal(err)
 	}
+	defer client.Close()
+	cleaner := gcscleaner.NewCleaner(client, cfg)
+	err = cleaner.DeleteOldBuckets(rootCtx)
+	if err != nil {
+		logrus.Warn(errors.Wrap(err, "deleting old buckets"))
```

Comment:
```suggestion
		logrus.Fatal(errors.Wrap(err, "while deleting old buckets"))
```

And remove `syscall`